analyses/molecular-subtyping-HGG/00-HGG-select-pathology-dx.Rmd

```diff
@@ -0,0 +1,185 @@
+In an initial round of subtyping, PNET specimens were reclassified (see [this comment on #609](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/609#issuecomment-602821376)).
+
+Let's take a look at the free text field.
+
+```{r}
+histologies_df %>% 
+  filter(short_histology == "HGAT") %>%
+  group_by(pathology_free_text_diagnosis) %>%
+  tally() %>%
+  arrange(desc(n))
+```
+
+As we might expect from a free text field, this is a lot less uniform.
+
+## Pathology diagnosis strings for inclusion
+
+These are the terms that we'll collapse together with `|` to detect strings in the `pathology_diagnosis` column.
+
+```{r}
+path_dx_terms <- c(
+  "High-grade glioma/astrocytoma",
+  "Diffuse Astrocytoma",
+  "Anaplastic Astrocytoma",
+  "Diffuse intrinsic pontine glioma",
+  "Diffuse midline glioma",
+  "Diffuse Astrocytoma",
+  "DIPG", 
+  "High grade glioma",
+  "Gliomatosis Cerebri"
+)
+```
+
+These are the terms that we'll collapse together with `|` to detect strings in the `pathology_free_text_diagnosis` column.
+
+```{r}
+free_text_dx_terms <- c(
+  "high grade glioma",
+  "high-grade glioma",
+  "hgg",
+  "anaplastic astrocytoma",
+  "glioblastoma",
+  "diffuse midline glioma",
+  "diffuse astrocytoma",
+  "dipg",
+  "gbm",
+  "anaplastic pilocytic astrocytoma",
+  "anaplastic gliomatosis cerebri",
+  "anaplastic pleomorphic xanthoastrocytoma"
+)
+```
+
+Let's take a look at a first attempt using these terms as described above.
+
+```{r}
+filtered_on_dx_df <- histologies_df %>%
+  filter(str_detect(pathology_diagnosis, 
+                    paste0(path_dx_terms, collapse = "|")) | 
+           str_detect(pathology_free_text_diagnosis, 
+                     paste0(free_text_dx_terms, collapse = "|"))) %>%
+  select(Kids_First_Biospecimen_ID, 
+         sample_id, 
+         Kids_First_Participant_ID,
+         pathology_diagnosis,
+         pathology_free_text_diagnosis,
+         integrated_diagnosis, 
+         short_histology)
+
+filtered_on_dx_df
+```
+
+Let's tally the values in `pathology_diagnosis` in this data frame.
+
+```{r}
+filtered_on_dx_df %>%
+  group_by(pathology_diagnosis) %>%
+  tally() %>%
+  arrange(desc(n))
+```
+
+Why are there low-grade glioma samples included?
+
+```{r}
+filtered_on_dx_df %>%
+  filter(pathology_diagnosis == "Low-grade glioma/astrocytoma (WHO grade I/II)")
+```
+
+We can exclude LGG samples on the basis of the values in `pathology_diagnosis`, too.
+
+```{r}
+exclude_path_dx_terms <- c("Low-grade glioma/astrocytoma")
+```
+
+### Save the strings we'll use downstream
+
+Create a list with the strings we'll use for inclusion and exclusion.
+
+```{r}
+terms_list <- list(include_path_dx = path_dx_terms,
+                   include_free_text = free_text_dx_terms,
+                   exclude_path_dx = exclude_path_dx_terms)
+```
+
+Save this list as JSON.
+
+```{r}
+writeLines(jsonlite::toJSON(terms_list, pretty = TRUE), output_file)
```

Comment:
I don't know why this isn't writing this with one list element per line, but that seems worth doing. It seems that `pretty=TRUE` does not work exactly the same way as `jsonlite::prettify()`, which does what I want:
```suggestion
writeLines(jsonlite::prettify(jsonlite::toJSON(terms_list)), output_file)
```